R/npm-scoring.R

```diff
@@ -7,6 +7,107 @@ PROTEIN_SCORE_THRESHOLDS <- c(8, 6.4, 4.8, 3.2, 1.6)
 NSP_FIBRE_SCORE_THRESHOLDS <- c(3.5, 2.8, 2.1, 1.4, 0.7)
 AOAC_FIBRE_SCORE_THRESHOLDS <- c(4.7, 3.7, 2.8, 1.9, 0.9)
 
+#' NPM scoring dispatcher
+#' 
+#' This function is written for calculating NPM scores across an entire row
+#' in a dataframe. It is wrapper logic for the underlying NPM_score_function 
+#' when applied to an NPMCalculator input data frame as such it is highly inflexible without 
+#' expected column names. It expects the specific gravity conversions to already have been performed
+#' and values outputted to a new single column.
+#' 
+#' @param row the row from an NPMCalculator dataframe
+#' @param sg_adjusted_label a character value specifying the name of the specific gravity adjusted column
+#' @return a matrix of NPM scores
+
+NPMScore <- function(row, sg_adjusted_label) {
```

Comment:
This function is pretty big, and unwieldy, but as a first cut of how to handle the presumed input data.frame organisation it appears to work as required.

Future questions are maybe:
- [ ] How do we break this up into smaller units
- [ ] How do we test this function
- [ ] How do we error handle this function